lib/note-detail.jsx

```diff
@@ -1,62 +1,166 @@
-import React, { PropTypes } from 'react';
-import marked from 'marked';
+import React, { Component, PropTypes } from 'react';
 import Textarea from 'react-textarea-autosize';
-import { noop, get, debounce } from 'lodash';
 import analytics from './analytics';
+import marked from 'marked';
+import { get, debounce, invoke } from 'lodash';
 import { viewExternalUrl } from './utils/url-utils';
 
-const uninitializedNoteEditor = { focus: noop };
 const saveDelay = 2000;
 
-export default React.createClass( {
+const isValidNote = note => note && note.id;
+const prependTab = l => `\t${ l }`;
+const removeLeadingTab = l => l && l.length && l[0] === '\t' ? l.substring( 1 ) : l;
 
-	propTypes: {
+export default class NoteDetail extends Component {
+	static propTypes = {
 		note: PropTypes.object,
 		previewingMarkdown: PropTypes.bool,
 		fontSize: PropTypes.number,
 		onChangeContent: PropTypes.func.isRequired
-	},
+	};
+
+	constructor( props ) {
+		super( props );
 
-	componentWillMount: function() {
+		this.storeNoteEditor = r => this.noteEditor = r;
 		this.queueNoteSave = debounce( this.saveNote, saveDelay );
-		this.noteEditor = uninitializedNoteEditor;
-	},
+	}
 
-	componentDidMount: function() {
+	componentDidMount = () => {
 		// Ensures note gets saved if user abruptly quits the app
 		window.addEventListener( 'beforeunload', this.queueNoteSave.flush );
-	},
-
-	initializeNoteEditor: function( noteEditor ) {
-		this.noteEditor = noteEditor;
-	},
-
-	isValidNote: function( note ) {
-		return note && note.id;
-	},
+		window.addEventListener( 'keydown', this.interceptTabPresses );
```

Comment:
@nfcampos the tab is a strange character on the web. the tab key doesn't trigger any of the React keypress events from the textarea because it's designed as a way to advance the input focus across the page. these behaviors we're adding actually break web accessibility in one sense because we're changing what the tab key does. on the other hand, this is more of an app than a webpage and the other apps already hold this behavior.

in short, no we couldn't do it with a React event because it won't fire when we need it